cmd/commands/image.go

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2018 The original author or authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package commands
+
+import (
+	"errors"
+	"github.com/projectriff/riff/pkg/core"
+	"github.com/spf13/cobra"
+)
+
+func Image() *cobra.Command {
+	return &cobra.Command{
+		Use:   "image",
+		Short: "Interact with docker images",
+	}
+}
+
+func ImageRelocate(c *core.Client) *cobra.Command {
+	options := core.RelocateImageOptions{}
+
+	command := &cobra.Command{
+		Use:   "relocate",
+		Short: "Relocate docker image names to another registry",
+		Long: `Relocate either a single YAML file or a riff manifest and its YAML files so that image names refer to
+another (private or public) registry.
+
+To transform a single YAML file, use the '--yaml' flag to specify the path or URL of the YAML file. Use the '--output'
+flag to specify the path to contain the transformed YAML file. If '--output' is an existing directory, the YAML file
+will be created in that directory. Otherwise the YAML file will be written to the path specified in '--output'.
+
+To transform a manifest, use the '--manifest' flag to specify the path of a manifest file which provides the paths or
+URLs of the YAML definitions of riff components. Use the '--output' flag to specify the path of a directory to contain
+the transformed manifest and YAML definitions.
+
+Specify the registry hostname using the '--registry' flag, the user owning the images using the '--registry-user' flag,
+and a complete list of the images to be mapped using the '--images' flag. The '--images' flag contains the path of animage manifest file with contents of the following form:
+
+    manifestVersion: 0.1
+    images:
+    ...
+    - istio/sidecar_injector
+    ...
+    - gcr.io/knative-releases/github.com/knative/serving/cmd/autoscaler@sha256:76222399addc02454db9837ea3ff54bae29849168586051a9d0180daa2c1a805
+    ...
+    
+`,
+		Example: `  riff image relocate --manifest=/path/to/manifest --registry=hostname --user=username --images=/path/to/image/manifest --output=/path/to/output/dir
+  riff image relocate --yaml=/path/to/yaml/file --registry=hostname --user=username --images=/path/to/image/manifest --output=/path/to/output`,
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// TODO: implement support for global flags - for now don't allow their use
```

Comment:
Those flags actually don't make sense for that command (as opposed to system install, but where we don't honor them). Should maybe revisit the whole PersistentFlags approach on rootCmd then

Reply:
Agreed. I'd like the revisiting of the PersistentFlags approach to be in a separate issue to avoid scope-creep here.

Reply:
Absolutely. I'm just saying that the (copy/pasted) comment is misleading